src/blob/index/core.rs

```diff
@@ -244,9 +244,21 @@ where
         }
     }
 
-    fn mark_all_as_deleted_in_memory(headers: &mut InMemoryIndex<K>, key: &K) -> Option<u64> {
+    fn mark_all_as_deleted_in_memory(
+        headers: &mut InMemoryIndex<K>,
+        key: &K,
+        default_header: RecordHeader,
+    ) -> Result<u64> {
         debug!("headers: {}", headers.len());
-        headers.remove(key).map(|v| v.len() as u64)
+        if let Some(v) = headers.get_mut(key) {
+            for h in v.iter_mut() {
+                h.mark_as_deleted()?
```

Comment:
This is not good. It is better to keep existing records as is and push deletion record to the end of vector